src/storage/wal_replayer.cpp

```diff
@@ -214,21 +221,34 @@ void WALReplayer::replayDropCatalogEntryRecord(const WALRecord& walRecord) {
     if (!(isCheckpoint && isRecovering)) {
         return;
     }
-    auto dropTableRecord =
+    auto dropEntryRecord =
         ku_dynamic_cast<const WALRecord&, const DropCatalogEntryRecord&>(walRecord);
-    auto tableID = dropTableRecord.tableID;
-    auto tableEntry =
-        clientContext.getCatalog()->getTableCatalogEntry(&DUMMY_READ_TRANSACTION, tableID);
-    switch (tableEntry->getTableType()) {
-    case TableType::NODE:
-    case TableType::REL:
-    case TableType::RDF: {
-        clientContext.getCatalog()->dropTableSchema(&DUMMY_WRITE_TRANSACTION, tableID);
-        // During recovery, storageManager does not exist.
-        if (clientContext.getStorageManager()) {
-            clientContext.getStorageManager()->dropTable(tableID);
+    auto entryID = dropEntryRecord.entryID;
+    switch (dropEntryRecord.entryType) {
+    case CatalogEntryType::NODE_TABLE_ENTRY:
+    case CatalogEntryType::REL_TABLE_ENTRY:
+    case CatalogEntryType::REL_GROUP_ENTRY:
+    case CatalogEntryType::RDF_GRAPH_ENTRY: {
+        auto tableEntry =
+            clientContext.getCatalog()->getTableCatalogEntry(&DUMMY_READ_TRANSACTION, entryID);
+        switch (tableEntry->getTableType()) {
```

Comment:
I don't think you need to switch on tableType here. Switch on `entryType` is fine.